### ISSUE TITLE
M1 MacOS support

### ISSUE THREAD

**Issue by zakir8** [soylent/weblink#8]
It needs macOS support. Ruby comes pre installed.

**Comment by soylent** [comment]
Hi, @zakir8 macOS is fully supported. To install weblink, you can open the terminal app and run `gem install weblink`. Then just follow the rest of the instructions in the README.

**Comment by zakir8** [comment]
Hi @soylent, I consistently get this error message. I have tried 'gem install weblink' in the user home directory. I also tried cloning the master branch and running `weblink.rb` file.I get the same error. Btw, I am using M1 Mac.


```Unable to load the EventMachine C extension; To use the pure-ruby reactor, require 'em/pure_ruby'
Traceback (most recent call last):
	2: from lib/weblink.rb:1:in `<main>'
	1: from /Users/zakir/.rbenv/versions/2.7.5/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'
/Users/zakir/.rbenv/versions/2.7.5/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require': cannot load such file -- em-websocket (LoadError)
	9: from lib/weblink.rb:1:in `<main>'
	8: from /Users/zakir/.rbenv/versions/2.7.5/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:149:in `require'
	7: from /Users/zakir/.rbenv/versions/2.7.5/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:160:in `rescue in require'
	6: from /Users/zakir/.rbenv/versions/2.7.5/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:160:in `require'
	5: from /Users/zakir/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/em-websocket-0.5.3/lib/em-websocket.rb:3:in `<top (required)>'
	4: from /Users/zakir/.rbenv/versions/2.7.5/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'
	3: from /Users/zakir/.rbenv/versions/2.7.5/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'
	2: from /Users/zakir/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:8:in `<top (required)>'
	1: from /Users/zakir/.rbenv/versions/2.7.5/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'
/Users/zakir/.rbenv/versions/2.7.5/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require': dlopen(/Users/zakir/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/eventmachine-1.2.7/lib/rubyeventmachine.bundle, 0x0009): symbol not found in flat namespace '_BIO_ctrl' - /Users/zakir/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/eventmachine-1.2.7/lib/rubyeventmachine.bundle (LoadError)
```

**Comment by zakir8** [comment]
Downgrading ruby version to `2.6.8` using rbenv worked for me. I tried using version `2.7.5` and `3.0.1` and somehow they threw that error.

**Comment by soylent** [comment]
Thank you, @zakir8

To fix the error on Ruby 2.7 and 3.0, you need to reinstall eventmachine with the following option:
```sh
gem install eventmachine -- --with-openssl-dir=~/.rbenv/versions/<version>/openssl/include/
```

weblink should work without any issues on Ruby 2.6 and 3.1.

**Comment by zakir8** [comment]
Perfect! installing event machine first before doing `gem install weblink` worked!
